sys/xtimer/xtimer_core.c

```diff
@@ -23,32 +23,42 @@
 #include "irq.h"
 
 /* WARNING! enabling this will have side effects and can lead to timer underflows. */
-#define ENABLE_DEBUG 0
+#define ENABLE_DEBUG (0)
 #include "debug.h"
 
-static volatile int _in_handler = 0;
+static int _in_handler = 0;
```

Comment:
I figured it is only changed to one, possibly read, changed to zero within the handler ISR. Outside it is only read and then should always be zero. So volatile doesn't protect anything. More atomic access is not needed.